expfmt/text_create.go

```diff
@@ -76,10 +76,6 @@ var (
 //
 // This method fulfills the type 'prometheus.encoder'.
 func MetricFamilyToText(out io.Writer, in *dto.MetricFamily) (written int, err error) {
-	// Fail-fast checks.
-	if len(in.Metric) == 0 {
-		return 0, fmt.Errorf("MetricFamily has no metrics: %s", in)
-	}
```

Comment:
Tests are failing because of this. I guess I would need to introduce a new function just to keep MetricFamilies without timeseries around?

Reply:
...or variadic option 🤔 or "both", so

NewMFConverter(opts).MetricFamilyToText

then keeping MetricFamilyToText as NewMFConverter(nil).MetricFamilyToText(....)